src/app/shared/directives/border-highliter.directive.ts

```diff
@@ -0,0 +1,30 @@
+import { Directive, ElementRef, Input } from '@angular/core';
+
+@Directive({
+  selector: '[crsBorderHighliter]'
+})
+export class BorderHighliterDirective {
+
+  @Input('crsBorderHighliter') date: Date;
+
+  constructor(private el: ElementRef) {}
+
+
+  ngOnInit() {
+      const today = new Date()
+      if (  this.date.getTime() < today.getTime() && 
+            this.date.getTime() >= this.addDays(today, -14).getTime()) {
+        this.el.nativeElement.style.border = '2px solid green';
```

Comment:
Angular docs recommends to not use `nativeElement` directly.

Here's the reason, Angular has different renderers depending on the platform we target. This `nativeElement` works in browsers, but would fail in case we try to render it using Server Side Rendering with Angular Universal.

Check here for `nativeElement` docs: https://angular.io/api/core/ElementRef#nativeElement

Check here how to user the `Renderer2`: https://angular.io/api/core/Renderer2

Also, I recommend to use CSS classes to switch between different styles.

Reply:
Done.
But I have a question. If I add classes instead of inline styles then I decouple directive functionality with the styling since css rule would be somewhere else, but not with directive itself. Is it ok? For now I put my css selectors and style in course css. But what the best approach here?

Reply:
You're right, @agorodnichev. For this Directive use case we should set the styles directly to the `Element` using `Renderer2` instead of appending a CSS class.

However, we don't need to limit ourselves only to this use case. For example, the `ngx-bootstrap` package, which is a port of Bootstrap components to Angular, uses the `Renderer2` to toggle CSS classes defined by the Bootstrap CSS framework. These classes are "global" CSS classes and are not scoped to a single element or component.

So, for this HW, use `Renderer2.setStyle`. But, throughout your projects, don't limit yourself on just this approach.

Reply:
Done.
Thanks for this explanation. Now I understand how all these packages work (Angular Material, Bootstrap, etc)